### PR TITLE
mmg : add variant to install private headers for parmmg packaged

### LINKED ISSUE #14: mmg: new package version up to 5.8
### Summary

new versions up to 5.8 are available

### Rationale

_No response_

### Description

_No response_

### Additional information

_No response_

### General information

- [x] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by prudhomm**
add private variant to install private headers thanks to MMG_INSTALL_PRIVATE_HEADERS needed by parmmg

/cc @jcortial-safran

- closes feelpp/spack#14

### CODE REVIEW

**Comment by jcortial-safran** [comment]
I would prefer a more explicit variant name, e.g. `private_headers`.

The variant should be also appear as a dependency in the parmmg package : https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/parmmg/package.py

**Comment by jcortial-safran** [comment]
Unless I am mistaken the option is only available for version 5.7.0 or later : https://github.com/MmgTools/mmg/commit/9f98ef4050a59b56e651e1641e07b2b1253917c0

 So the option should not be available for versions earlier than 5.7.0.

**Review by tldahlgren — DISMISSED** [review]
LGTM.  @jcortial-safran ?

**Comment by jcortial-safran** [comment]
> LGTM. @jcortial-safran ?

I have suggested a couple of minor changes.

**Comment by tldahlgren** [comment]
Ping me with a comment here when this is ready to merge.

**Comment by prudhomm** [comment]
@tldahlgren @jcortial-safran I believe I applied the required changes.

File: var/spack/repos/builtin/packages/mmg/package.py [hunk 1]
```diff
@@ -60,10 +60,7 @@ class Mmg(CMakePackage):
 
 class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
     def cmake_args(self):
-        shared_active = self.spec.satisfies("+shared")
```
Comment (jcortial-safran) [inline var/spack/repos/builtin/packages/mmg/package.py:61]: This line should not have been deleted. Please restore it, currently the recipe is broken.

File: var/spack/repos/builtin/packages/mmg/package.py [hunk 2]
```diff
@@ -50,6 +51,7 @@ class Mmg(CMakePackage):
     variant("scotch", default=True, description="Enable SCOTCH library support")
     variant("doc", default=False, description="Build documentation")
     variant("vtk", default=False, when="@5.5.0:", description="Enable VTK I/O support")
+    variant("private_headers", default=True, description="Enable private headers")
```
Comment (jcortial-safran) [inline var/spack/repos/builtin/packages/mmg/package.py]: I would prefer the default value to be `False`.

**Review by jcortial-safran — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/mmg/package.py [hunk 3]
```diff
@@ -72,6 +73,12 @@ def cmake_args(self):
             self.define("LIBMMGS_STATIC", not shared_active),
             self.define("LIBMMG_STATIC", not shared_active),
         ]
+        # Add conditional flag for MMG_INSTALL_PRIVATE_HEADERS
+        if self.spec.satisfies("@5.7.0:"):
+            args.append(
+                self.define("MMG_INSTALL_PRIVATE_HEADERS", self.spec.satisfies("+private_headers"))
+            )
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/mmg/package.py]: If this option only applies for versions `@5.7:`, then you could pass that as a `when` argument to the variant then not bother with the check here.

**Comment by jcortial-safran** [comment]
LGTM @tldahlgren

**Review by jcortial-safran — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**